### PR TITLE
sg: Build ./dev/sg on Buildkite, don't run all tests for sg changes

### PR DESCRIPTION
**Pull request by mrnugget**

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
🆒

File: enterprise/dev/ci/internal/ci/pipeline.go [hunk 1]
```diff
@@ -98,6 +98,13 @@ func GeneratePipeline(c Config) (*bk.Pipeline, error) {
 			addPostgresBackcompat,         // ~0.25m
 		}
 
+	case c.isSgOnly():
```
Comment (keegancsmith) [inline enterprise/dev/ci/internal/ci/pipeline.go:101]: need to check it is a PR. We don't want to mark main as green for these changes if it is currently red.

**Review by keegancsmith — APPROVED** [review]
Sad that go tests have come to this. How badly do you need this extra special casing complexity vs waiting a little longer?

**Comment by mrnugget** [comment]
> Sad that go tests have come to this. How badly do you need this extra special casing complexity vs waiting a little longer?

Sad about what? I don't follow.

Without this special casing _everything_ runs when I change some files in `./dev/sg`. Look at this build for the last PR that only changed `sg` stuff: `sg.config.yaml` and `./dev/sg/*.go`. https://buildkite.com/sourcegraph/sourcegraph/builds/97856 TypeScript linting, Puppeteer, yarn test, browser extension.

**Comment by mrnugget** [comment]
In case you meant `go-test-sg.sh`: just removed it and run the full `go-test.sh` (which seems to have grown organically...)

**Comment by keegancsmith** [comment]
Oh I see, it isn't just about go files you have other files as well. That makes sense. Ship it. Sadness was I thought you didn't want to rely on just go-test because you thought it was too slow.

**Comment by mrnugget** [comment]
> Sadness was I thought you didn't want to rely on just go-test because you thought it was too slow.

It _is_ too slow :stuck_out_tongue:  But to make you happy, I removed the additional file.

**Pull request merged**